pkg/matcher/README.md

```diff
@@ -42,18 +44,30 @@ Short Syntax
      <expr>      ::= any string
 ```
 
-**Tip**: In short syntax, for example, you can enable the glob format by starting the string with a `*`, while in the long syntax you need to define it more explicitly. The following examples are identical. `simple_patterns` can be used **only** with the long syntax.
+**Remember**: In short syntax, for example, you can enable the glob format by starting the string with a `*`, while in the long syntax you need to define it more explicitly. The following examples are identical. `simple_patterns` can be used **only** with the long syntax.
 
-Short Syntax: `'* * '`
-Long Syntax: `'glob':'*'`
+Examples:
+ - Short Syntax: `'* * '`
+ - Long Syntax: `'glob':'*'`
 
 ### String matcher
 The string matcher reports whether the given value equals to the string ( use = ).
 
 Examples: 
- - `'== foo'` will match only if the string is `'foo'`.
- - `'[!] == bar'` will match any string that is not `'bar'`.
+ - `'= foo'` will match only if the string is `'foo'`.
+ - `'! = bar'` will match any string that is not `'bar'`.
+
+String matcher means **exact match** of the `string` with the `pattern`. There are other string match related cases:
```

Comment:
> String matcher means **exact match** of the `string` with the `pattern`

in that case it is not `pattern` but an exact value

Reply:
I would agree that this is confusing if we're saying the string matcher looks for exact matches, then also say it's a pattern. 

I actually would recommend we move this glob content down into the next section about globs as some additional examples of how to capture strings using globs.